### PR TITLE
Home work with file

### PR DESCRIPTION
**Pull request by 4Vitalii5**
Implemented logic for method getStatistic.
Added a methods :
- writeToFile (write array of text to file)
- readFromFile ( read from file to array of strings)
- getUniqueWords (returns array of strings with unique words)
- getOdd ( returns the odds of the elements)
- getSumItems (returns the sum of identical elements)
- getSortedData (returns sorted array from getSumItems by numeric value in descending order)

### CODE REVIEW

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 1]
```diff
@@ -1,7 +1,98 @@
...
+    private String[] getUniqueWords(String[] dataFromFile) {
+        StringBuilder uniqueArray = new StringBuilder();
+        String allStrings = Arrays.toString(dataFromFile).replaceAll("[^a-zA-Z ]+", "");
+        String[] strings = allStrings.split(" ");
+        for (String temp : strings) {
+            if (!uniqueArray.toString().contains(temp)) {
+                uniqueArray.append(temp).append(" ");
+            }
+        }
+        return String.valueOf(uniqueArray).split(" ");
```
[the remark this answers is not in the document]
Comment (4Vitalii5) [reply src/main/java/core/basesyntax/WorkWithFile.java:75]: I don't no how to use Set now. Maybe later I will use it

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 2]
```diff
@@ -1,7 +1,95 @@
...
+import java.io.BufferedWriter;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Arrays;
+
 public class WorkWithFile {
     public void getStatistic(String fromFileName, String toFileName) {
+        String[] data = getSortedData(getSumItems(fromFileName));
```
[the remark this answers is not in the document]
Comment (4Vitalii5) [reply src/main/java/core/basesyntax/WorkWithFile.java]: I know how to squeeze in three methods only with the use of the Collections. But I haven't actually taught them yet. If you know how to do it without using collections, please let me know.

**Comment by 4Vitalii5** [comment]
I know how to squeeze in three methods only with the use of the Collections. But I haven't actually taught them yet. If you know how to do it without using collections, please let me know.

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 3]
```diff
@@ -1,7 +1,56 @@
...
+
     public void getStatistic(String fromFileName, String toFileName) {
+        String[] textFromFile = readFromFile(fromFileName);
+        String report = generateReport(textFromFile);
+        writeToFile(report, toFileName);
+    }
+
+    private String[] readFromFile(String fromFileName) {
+        try {
+            return Files.readString(Path.of(fromFileName)).split("\r?\n");
```
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/WorkWithFile.java:20]: Files not supported closing resources automatically
It's better to use BufferedReader and try with resources

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 4]
```diff
@@ -1,7 +1,56 @@
...
+    }
+
+    private String generateReport(String[] data) {
+        StringBuilder report = new StringBuilder();
+        int supplySum = 0;
+        int buySum = 0;
+        int result;
+        for (String product : data) {
+            String[] line = product.split(COMMA);
+            if (line[0].equals("supply")) {
```
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/WorkWithFile.java:33]: supply should be constant

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 5]
```diff
@@ -1,7 +1,56 @@
...
+                supplySum += Integer.parseInt(line[1]);
+            } else {
+                buySum += Integer.parseInt(line[1]);
+            }
+        }
+        result = supplySum - buySum;
+        report.append("supply,").append(supplySum).append(System.lineSeparator());
```
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/WorkWithFile.java:40]: use constant for supply, buy, result and comma

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]

**Review by Elena-Bruyako — APPROVED** [review]

**Pull request closed without merging**